tests/frontier/opcodes/test_dup.py

```diff
@@ -59,9 +62,20 @@ def test_dup(
     account_code += dup_opcode
 
     # Save each stack value into different keys in storage
-    account_code += b"".join([Op.PUSH1(i) + Op.SSTORE for i in range(0x11)])
-
-    pre[account] = Account(code=account_code)
+    account_code += b"".join([Op.PUSH1(i) + Op.SSTORE for i in range(0x11)]) + Op.STOP
+
+    pre[account] = Account(
+        code=account_code
+        if evm_code_type == EVMCodeType.LEGACY
+        else Container(
+            sections=[
+                Section.Code(
+                    account_code,
+                    max_stack_height=19,
+                ),
+            ]
+        )
```

Comment:
To make this easier to use I think we need a standard method for this, like `maybe_eof(<code>, evm_code_type)` 

But it's not that simple as spit out raw code for legacy and for EOF wrap it in a container with  the code in section 0. 
 We also need to validate there are no forbidden opcodes (like jumps or calls ;) ) and calculate the stack height.

Reply:
This is where `pre.deploy_contract` from #584 will potentially come into play.

With this standardized way of contract deployment, the pre-allocation could try to convert the code into EOF automatically in many of the tests, and give the tester the option to implement the container manually if not possible. I.e. you can add the marker and then run the test, which should fail if the EOF container could not be produced, and then you need to manually implement the container.

This is one example for one of the updated EOF tests in that PR: https://github.com/ethereum/execution-spec-tests/pull/584/files#diff-e89eb58de375df2e5b5e0472312229ce973d61cddac4955c037bfe63ecccac48

Reply:
That will work great with yul honestly.  Just re-compile with the eof target when EOF lands in solidity.

Perhaps if we had configs for the types to test. Default value is fork dependent, (legacy pre Prague, both post), fails code generation if it cannot meet the flags), maybe a flag to error and the default is a warning.

Reply:
I added a `--evm-code-type` flag to `fill` and it's pretty interesting to see what tests contain code that is simply plug-and-play into EOF.

I'm currently analyzing the tests but if you pick up this branch and then do:
```bash
fill --fork=CancunEIP7692 --evm-code-type=eof_v1 --evm-bin=evmone-t8n ./tests/<some test>
```
you can start trying out tests from previous forks in EOF mode to see which ones we could mark as interesting to also fill in EOF containers.

So what we could do is:
- Use the `--evm-code-type` to manually find interesting pre-EOF tests that we should add the permanent markers to.
- Fix tests that are not EOF compatible but we would like to run them in EOF mode because they test interesting things
- Add either `@pytest.mark.with_all_evm_code_types` or `@pytest.mark.with_all_call_opcodes` to the appropriate tests.

Reply:
Tagging ipsilon for visibility @pdobacz @gumb0 @chfast

Reply:
I have been following along more or less. This is great stuff, I'll give it a spin right away, thank you!

Reply:
I've quickly experimented with one test (MCOPY) here https://github.com/ethereum/execution-spec-tests/pull/726. I think it works OK. I'm not yet sure myself if it is the right direction, but posting here to gather thoughts and feedback. Let me know what you think @marioevz. I'll also experiment more with this tmr.

Overall (being conscious that the MCOPY change is actually quite involved) I think the marker works great! (I haven't done much of the `call_opcode` one yet, this one is more quirky)

(I also tried the 7702 tests, but that got stuck on 7702 being still in flux, so I left it for after 7702 stabilizes and implementations are more ready)

Also this gets me wondering: any remote chance we have sth similar for `ethereum/tests`? :grimacing:

Reply:
> (I also tried the 7702 tests, but that got stuck on 7702 being still in flux, so I left it for after 7702 stabilizes and implementations are more ready)

Maybe this one is due to evmone not supporting 7702 yet? It might be possible to do this with besu, since Prague has EOF activated.

> Also this gets me wondering: any remote chance we have sth similar for `ethereum/tests`? 😬

I think it would be great, but it would be extremely time consuming IMO. It might be easier to just start porting tests over. Dimitry has already been doing this for a bit now.

Reply:
I still think it would yield almost no usable tests because I have the feeling most of them have either jumps or use invalid EOF opcodes.

One way of trying out this theory could be to use `evm_bytes_to_python` to load each account contract in every `pre` of the filled fixtures from `ethereum/tests`, and see what percentage can be put into `Container.Code` and results in valid EOF code.

Reply:
> I still think it would yield almost no usable tests because I have the feeling most of them have either jumps or use invalid EOF opcodes.

Ok, that indeed is problematic (unless we treat those as usable tests for EOF validation - however those would be rather trivial cases of undefined opcode). I'll make a self note to do the experiment you describe. Thanks for the input!